apps/prairielearn/src/ee/lib/ai-grading/ai-grading.ts

```diff
@@ -145,9 +146,23 @@ export async function aiGrade({
     });
     job.info(`Found ${instance_questions.length} submissions to grade!`);
 
+    const gradingJobMapping = await selectGradingJobsInfo(instance_questions);
+
     let error_count = 0;
     // Grade each instance question
     for (const instance_question of instance_questions) {
+      const grading_jobs = gradingJobMapping[instance_question.id] ?? [];
+
+      const manualGradingJob = grading_jobs.find((job) => job.grading_method === 'Manual');
+      const aiGradingJob = grading_jobs.find((job) => job.grading_method === 'AI');
+
+      const should_update_score =
+        !manualGradingJob || // no manual grading job
+        (manualGradingJob &&
+          aiGradingJob &&
+          manualGradingJob.graded_at &&
+          aiGradingJob.graded_at &&
+          manualGradingJob.graded_at < aiGradingJob.graded_at); // exist more recent ai grading job
```

Comment:
Help me think through this: won't think only prevent the first AI grading run to follow a human manual grade from updating the score? That is, consider the case where a human grades a submission, then we AI grade the same one. If we then try to AI-grade it again, won't `aiGradingJob.graded_at` be greater than `manualGradingJob.graded_at`, and then we'll update the score?

If this is wrong, let me know what I'm missing, and consider comments to clarify what is happening here and how.